src/utils/useProjectId.ts

```diff
@@ -0,0 +1,25 @@
+import { useChannel } from "@storybook/manager-api";
+import React from "react";
+
+import { UPDATE_PROJECT, UpdateProjectPayload } from "../constants";
+
+const { CHROMATIC_PROJECT_ID } = process.env;
+
+export const useProjectId = (): [
+  projectId: string,
+  updateProject: (projectId: string, projectToken?: string) => void,
+  projectIdChanged: boolean,
+  clearProjectIdChanged: () => void
+] => {
+  const [projectId, setProjectId] = React.useState<string | null>(CHROMATIC_PROJECT_ID);
```

Comment:
Instead of using the env variable here we want to emit an event, and "await" the response.
That way we always get the fresh value, not what it was when storybook started.

Reply:
I commented about this in the PR description: https://github.com/chromaui/addon-visual-tests/pull/10#issuecomment-1633606915

Do you agree this is OK for now and we can iterate as we start polling the branch/commit info (and thus need some events to update the manager w/ new info)?

Reply:
Yeah, it's OK with the caveat that you'll need to restart storybook after setting the projectId.
That's OK for internal use, I'd say.